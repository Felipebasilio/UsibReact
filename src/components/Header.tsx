import React from 'react';
import { NavLink } from 'react-router-dom';


const Header = () => {
  return (
    <header className="site-header py-1 sticky-top my-0">
      <nav className="navbar navbar-expand navbar-dark d-flex justify-content-start">

          <NavLink to="/" className="px-5"  activeClassName="active" exact>
            USIB
          </NavLink>
          <NavLink to="/listQuotes" className="px-4"  activeClassName="active" exact>
            Or√ßamentos
          </NavLink>

      </nav>
    </header>
  );
};

export default Header;