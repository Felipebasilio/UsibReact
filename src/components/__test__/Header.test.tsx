import ReactDOM from 'react-dom';
import Header from '../Header';
import renderer from 'react-test-renderer'
import { Router } from 'react-router-dom';

import { createBrowserHistory } from 'history';

const history = createBrowserHistory();

it('renders Header without crashing', () => {
    let div = document.createElement('div');
    ReactDOM.render( <Router history={ history }><Header/></Router>, div );
    ReactDOM.unmountComponentAtNode(div);
    
})


it('Matches snapshot', () => {
    const HeaderComponnet = renderer.create(
        <Router history={history}>
            <Header />
        </Router>
    ).toJSON();
            
    expect(HeaderComponnet).toMatchSnapshot();
})

