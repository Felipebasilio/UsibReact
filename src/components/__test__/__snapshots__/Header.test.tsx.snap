// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<header
  className="site-header py-1 sticky-top my-0"
>
  <nav
    className="navbar navbar-expand navbar-dark d-flex justify-content-start"
  >
    <a
      aria-current="page"
      className="px-5 active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      USIB
    </a>
    <a
      aria-current={null}
      className="px-4"
      href="/listQuotes"
      onClick={[Function]}
    >
      Or√ßamentos
    </a>
  </nav>
</header>
`;
